usr/share/rear/restore/FDRUPSTREAM/default/260_copy_log_and_report.sh

```diff
@@ -1,14 +1,44 @@
-# upstream.log and upstream.rpt on the ReaR system contains valuable 
+# upstream.log and upstream.rpt on the ReaR system contain valuable 
 # information about the disaster recovery, so we archive *.log and *.rpt
 # to the restored system.
 
-EXTENSION="restore_$( date +%F_%T )"
+PREFIX="rear_$( date +%F_%T_%N )"
+SERVICENAME=( $(ps -ef | grep uscmd1 | grep servicename | grep -Po 'servicename=\K[^"]+') )
```

Comment:
I think when SERVICENAME is empty
the `for dir in "${SERVICENAME[@]}"; do` loop
is not at all run which means that script does nothing.

In contrast the previous script
https://github.com/rear/rear/blob/master/usr/share/rear/restore/FDRUPSTREAM/default/260_copy_log_and_report.sh
did not depend on a non-empty SERVICENAME

Reply:
I can put in a check to see if SERVICENAME is empty, and if so exit this script with an error.  This would indicate that FDR/Upstream stopped/crashed after performing the restore.  ReaR should continue in that case, but the user would be notified that log files have not been copied to the recovered system.  

That brings up another question:  where can I find an explanation of LogPrint, LogPrintError, and LogPrintIfError?

Reply:
I found usr/share/rear/lib/_input-output-functions.sh has explanations of the functions I was looking for.